core/src/main/java/org/operationcrypto/hivej/communication/ConnectionManager.java

```diff
@@ -0,0 +1,153 @@
+/*
+ *     This file is part of HiveJ
+ * 
+ *     HiveJ is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * 
+ *     HiveJ is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ * 
+ *     You should have received a copy of the GNU General Public License
+ *     along with HiveJ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.operationcrypto.hivej.communication;
+
+import java.io.IOException;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class manages connections. Pools LoadBalance
+ * 
+ * @author <a href="https://github.com/marvin-we">marvin-we</a>
+ */
+public class ConnectionManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionManager.class);
+
+    /** The one and only instance. */
+    private static volatile ConnectionManager sConnectionManager;
+
+    /** A pointer to the next client to use. */
+    private int nextClient;
+    /** The list of endpoints known to the ConnectionManager. */
+    private CopyOnWriteArrayList<AbstractClient> clients;
```

Comment:
we may think about starting member variables with prefix m?

Reply:
This is the default SQ rule for member variables https://rules.sonarsource.com/java/tag/convention/RSPEC-116 whose regex does not enforce this by default.

I still see the benefit of it (and we should agree on something for sure :P) - So should I update the rule?

Reply:
i dont't mind - as it is less effort lets stick to the already configured rule.